components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -432,41 +433,37 @@ void DispatchBlockedEvent(const GURL& request_url,
 #endif
 }
 
-bool ShouldCleanReferrerForTopLevelNavigation(const std::string& method,
-                                              const GURL& referrer_url,
-                                              const GURL& request_url) {
-  // See https://github.com/brave/brave-browser/issues/8696
-  return ((method == "GET" || method == "HEAD") &&
-      !net::registry_controlled_domains::SameDomainOrHost(
-          referrer_url, request_url,
-          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES));
+bool IsSameOriginNavigation(const GURL& referrer, const GURL& target_url) {
+  const url::Origin original_referrer = url::Origin::Create(referrer);
+  const url::Origin target_origin = url::Origin::Create(target_url);
+
+  return original_referrer.IsSameOriginWith(target_origin);
 }
 
 bool MaybeChangeReferrer(
```

Comment:
Do we still need this? Im not aware of cases we'd be making any changes to the referrer, if we're just doing what chrome+flags says.  Are there other cases?  If no, can we just remove?

Reply:
We change the referrer - we do capping (in this function), while chrome does not. We could achieve capping via the feature flag, but I'm a bit scaried by the possibility of them ditching the flag. Also, our implementation of capping is shields-controlled.

Reply:
Gotcha, thanks @iefremov !